tools/eubfr-cli/commands/delete.js

```diff
@@ -0,0 +1,102 @@
+const url = require('url');
+const path = require('path');
+
+const aws4 = require('aws4');
+const dotenv = require('dotenv');
+const elasticsearch = require('elasticsearch');
+const request = require('request-promise-native');
+
+const getServiceLocation = require('../lib/getServiceLocation');
+const extractCredentials = require('../lib/extractCredentials');
+
+dotenv.config({
+  path: path.resolve(getServiceLocation('storage-deleter'), '.env'),
+});
+
+/**
+ * Deletes files from S3 for a given producer.
+ *
+ * @param {Array} files
+ *   List of files to delete.
+ * @param {Array} credentials
+ *   List of credentials for the producers.
+ */
+const deleteCommand = async ({ files, credentials }) => {
+  if (!process.env.DELETER_API) {
+    return console.error(
+      "DELETER_API environment variable is missing. Please redeploy by running 'yarn deploy' from project root"
+    );
+  }
+
+  // Prepare variables for further requests.
+  const resource = 'storage/delete';
+  const api = url.parse(`https://${process.env.DELETER_API}`);
+  const uri = `https://${process.env.DELETER_API}/${resource}`;
+  const resourcePath = `${api.path}/${resource}`;
+
+  const index = `${process.env.REACT_APP_STAGE}-meta`;
+  const host = `https://${process.env.REACT_APP_ES_PRIVATE_ENDPOINT}`;
+
+  const client = elasticsearch.Client({
+    host,
+    log: 'warning',
+    apiVersion: '6.2',
+  });
+
+  // Reuse throughout the two delete approaches.
+  const deleteFile = async (computedKey, creds) => {
+    try {
+      const params = {
+        uri,
+        host: api.host,
+        path: resourcePath,
+        headers: {
+          'x-amz-meta-computed-key': computedKey,
+        },
+      };
+
+      console.log(`Deleting ${computedKey} ...`);
+      await request.get(aws4.sign(params, creds));
+      return console.log(`${computedKey} has been deleted`);
+    } catch (e) {
+      return console.error(e);
+    }
+  };
+
+  // Marker to delete all files for all producers
+  if (files.length === 0) {
+    return credentials.forEach(async producer => {
```

Comment:
Beware of async + forEach!
https://codeburst.io/javascript-async-await-with-foreach-b6ba62bbf404